### PR TITLE
split transaction log and action log

### PR DESCRIPTION
**Pull request by CoderZhi**

### CODE REVIEW

File: action/receipt.go [hunk 1]
```diff
@@ -119,16 +115,34 @@ func (receipt *Receipt) Hash() hash.Hash256 {
 	return hash.Hash256b(data)
 }
 
+// Logs returns the list of transaction logs stored in receipt
```
Comment (yutongp) [inline action/receipt.go:118]: Wrong comment

File: action/protocol/staking/handlers.go [hunk 2]
```diff
@@ -121,18 +121,14 @@ func (p *Protocol) handleCreateStake(ctx context.Context, act *action.CreateStak
 	log.SetData(byteutil.Uint64ToBytesBigEndian(bucketIdx))
 
 	// generate create amount log
-	cLog := action.Log{
-		Address:     p.addr.String(),
-		BlockHeight: blkCtx.BlockHeight,
-		ActionHash:  actionCtx.ActionHash,
-		TransactionData: &action.TransactionLog{
+	return log, []*action.TransactionLog{
```
Comment (dustinxie) [inline action/protocol/staking/handlers.go:124]: prefer to keep cLog :=

File: action/protocol/staking/handlers.go [hunk 3]
```diff
@@ -263,19 +259,14 @@ func (p *Protocol) handleWithdrawStake(ctx context.Context, act *action.Withdraw
 		log.SetData(bucket.StakedAmount.Bytes())
 	}
 
-	// generate withdraw amount log
```
Comment (dustinxie) [inline action/protocol/staking/handlers.go:266]: keep the comment

File: action/receipt.go [hunk 4]
```diff
@@ -119,16 +115,34 @@ func (receipt *Receipt) Hash() hash.Hash256 {
...
+func (receipt *Receipt) Logs() []*Log {
+	return receipt.logs
+}
+
 // AddLogs add log to receipt and filter out nil log.
-func (receipt *Receipt) AddLogs(logs ...*Log) {
-	if receipt.Logs == nil {
-		receipt.Logs = make([]*Log, 0, len(logs))
```
Comment (dustinxie) [inline action/receipt.go:125]: should keep this?
saw places construct r = &action.Receipt{ ... } with logs = nil

File: action/receipt.go [hunk 5]
```diff
@@ -119,16 +115,34 @@ func (receipt *Receipt) Hash() hash.Hash256 {
...
+}
+
+// TransactionLogs returns the list of transaction logs stored in receipt
+func (receipt *Receipt) TransactionLogs() []*TransactionLog {
+	return receipt.transactionLogs
+}
+
+// AddTransactionLogs add transaction logs to receipt and filter out nil log.
+func (receipt *Receipt) AddTransactionLogs(logs ...*TransactionLog) *Receipt {
 	for _, l := range logs {
```
Comment (dustinxie) [inline action/receipt.go:140]: should check nil and receipt.transactionLogs = make()

**Pull request merged**